0022-circuit-block.md

```diff
@@ -0,0 +1,300 @@
+* the QPY API could be extended to retrieve the serialisers/deserialisers from a `backend` object, rather than needing a user to construct it themselves.
+
+If we _truly_ associated the Qiskit object `Backend` with a "quantum computer" (as opposed to `Target`'s QPU), we might even consider an API where the primitives entry points are moved onto the `BackendV3` object, as in:
+
+```python
+class BackendV3:
+    @property
+    @abstractmethod
+    def target(self) -> Target: ...
+    @abstractmethod
+    def allowed_annotations(self) -> AnnotationChecker: ...
+    @abstractmethod
+    def sample(self, sampler_pub_likes) -> SamplerJob: ...
+    @abstractmethod
+    def estimate(self, estimator_pub_likes) -> EstimatorJob: ...
+```
+
+This might naturally extend to such a `BackendV3` providing the additional information that is needed for Qiskit SDK to safely transpile entire sampler and estimator pubs; the compilation of a pub _also_ implies further processing will be done by a quantum computer, just like an annotation remaining in a single circuit.
+
+
+## Case study: twirling, noise learning, and mitigation with `qiskit-ibm-runtime`
+
+We envision two main workflows that users will follow once `Box`es become available, one for regular users and another one for power users:
+
+1. The workflow for regular users:
+    - Users initialize a `QuantumCircuit` without boxes, as they do today.
+    - They apply all of the desired transpiler passes, for example to map the circuit to an ISA circuit for the backend that they wish to use.
+    - They use (a convenience method built around) a new transpiler pass that collects the circuit's gates into boxes, with the ability of specifying different collection strategies.
+    - They submit their job.
+2. The workflow for power users:
+    - Users initialize a `QuantumCircuit` adding boxes manually as they wish.
+    - They submit their job.
+
+Setting up these two workflows presents notable advantages, such as:
+- All of the existing transpiler passes can be asked (at least initially) to simply ignore the boxes, since transpilation is meant to happen before the boxes are generated. This can save quite a lot of time that would otherwise be spent adding logic to every single transpiler pass -- and yet, we can choose to do this in the future.
+- Users of the primitives can inspect the boxes before submitting their jobs. This is a big improvement: today, the box generation happens in the server and can be unintuitive.
+
+In addition to supporting the two workflow above, we believe that the following workflow should *not* be supported:
+
+3. The workflow for "lazy" users:
+    - Users initialize a `QuantumCircuit` without boxes, as they do today.
+    - They apply all of the desired transiler passes, for example to map the circuit to an ISA circuit for the backend that they wish to use.
+    - They submit their job.
+    - The server creates boxes for the users "behind the scenes."
```

Comment:
while important, this comment i think has now strayed far from the goal of this RFC since it's about the runtime service implementation and features rather than boxes and annotations. we could instead highlight that boxes bring the possibility of making such operations transparent while still allowing the tight sampling loops to happen near-time